src/app/dataset/statistics/statistics.component.ts

```diff
@@ -16,62 +17,82 @@ export class StatisticsComponent implements OnInit {
   expandedStatistics = false;
   isLoading = false;
   notification?: Notification;
-  statistics: NodeStatistics[];
+  statistics: Statistics;
+  taskId?: string;
 
   ngOnInit(): void {
     this.loadStatistics();
   }
 
+  setLoading(loading: boolean): void {
+    this.isLoading = loading;
+  }
+
   // load the data on statistics and display this in a card (=readonly editor)
   loadStatistics(): void {
+    this.setLoading(true);
     this.workflows.getFinishedDatasetExecutions(this.datasetData.datasetId, 0).subscribe(
       (result) => {
-        let taskId: string | undefined;
         if (result.results.length > 0) {
           // find validation in the latest run, and if available, find taskid
           for (let i = 0; i < result.results[0].metisPlugins.length; i++) {
             if (result.results[0].metisPlugins[i].pluginType === 'VALIDATION_EXTERNAL') {
-              taskId = result.results[0].metisPlugins[i].externalTaskId;
+              this.taskId = result.results[0].metisPlugins[i].externalTaskId;
             }
           }
         }
-
-        if (!taskId) {
+        if (!this.taskId) {
           return;
         }
-        this.isLoading = true;
-        this.workflows.getStatistics('validation', taskId).subscribe(
+        this.workflows.getStatistics('validation', this.taskId).subscribe(
           (resultStatistics) => {
-            let statistics = resultStatistics.nodeStatistics;
-
-            if (statistics.length > 100) {
-              statistics = statistics.slice(0, 100);
-            }
-            statistics.forEach((statistic) => {
-              const attrs = statistic.attributesStatistics;
-              if (attrs.length > 100) {
-                statistic.attributesStatistics = attrs.slice(0, 100);
-              }
-            });
-
+            const statistics = resultStatistics;
             this.statistics = statistics;
-            this.isLoading = false;
+            this.setLoading(false);
           },
           (err: HttpErrorResponse) => {
             const error = this.errors.handleError(err);
             this.notification = httpErrorNotification(error);
-            this.isLoading = false;
+            this.setLoading(false);
           },
         );
-        return;
       },
       (err: HttpErrorResponse) => {
         const error = this.errors.handleError(err);
         this.notification = httpErrorNotification(error);
+        this.setLoading(false);
       },
     );
   }
 
+  loadMoreAttrs(xPath: string): void {
+    if (!this.taskId) {
```

Comment:
Does it make sense to do this check in the template itself? So that there is no button at all if there is no taskId?